src/System.Reflection.Emit.ILGeneration/tests/CustomAttributeBuilderTests.cs

```diff
@@ -644,32 +647,29 @@ public void NullValueForPrimitiveTypeInConstructorArgs_ThrowsNullReferenceExcept
             ConstructorInfo con = typeof(TestAttribute).GetConstructor(new Type[] { typeof(int) });
             object[] constructorArgs = new object[] { null };
 
-            Assert.Throws<NullReferenceException>(() => new CustomAttributeBuilder(con, constructorArgs));
-            Assert.Throws<NullReferenceException>(() => new CustomAttributeBuilder(con, constructorArgs, new FieldInfo[0], new object[0]));
-            Assert.Throws<NullReferenceException>(() => new CustomAttributeBuilder(con, constructorArgs, new PropertyInfo[0], new object[0]));
-            Assert.Throws<NullReferenceException>(() => new CustomAttributeBuilder(con, constructorArgs, new PropertyInfo[0], new object[0], new FieldInfo[0], new object[0]));
```

Comment:
If we find this breaks some scenario, we may have to change this back to a NullReferenceException (though via an explicit throw with a comment, rather than accidentally as was the case before.)